install-deps.sh

```diff
@@ -433,7 +433,11 @@ find . -name tox.ini | while read ini ; do
         fi
     )
 done
+curl -O https://bootstrap.pypa.io/get-pip.py
+$(SUDO) python get-pip.py
+$(SUDO) pip install pyopenssl
```

Comment:
python-openssl is installed as a dependency of ceph-mgr. so presumably it should be around after running `install-deps.sh`. could you pastebin the error message if pyopenssl is not installed with `pip`?

Reply:
HEALTH_WARN Module 'restful' has failed dependency: No module named OpenSSL
MGR_MODULE_DEPENDENCY Module 'restful' has failed dependency: No module named OpenSSL
    Module 'restful' has failed dependency: No module named OpenSSL

Reply:
ah! i see. `python-openssl` is listed as a runtime dependency but not a *build*-time dependency. that's why it is not installed by `install-deps.sh`! see @tserong 's analysis at https://tracker.ceph.com/issues/37584#note-13 .

so this PR is actually fixing https://tracker.ceph.com/issues/37584 . i am proposing an alternative to address this issue at https://tracker.ceph.com/issues/37584#note-15. and a fix is posted at #26402